src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -171,6 +171,10 @@ else if (!hasText(leftCurly)) {
                     ast.getText());
             }
         }
+
+        if (ast.getType() == TokenTypes.LITERAL_CASE) {
+            checkForEmptyCase(ast);
+        }
```

Comment:
There were cyclomatic complexity issues (exceed the default max by 1 ) even when I used a single separate function. Hence I used 3 separate functions.